include/dlaf/matrix/matrix.h

```diff
@@ -56,10 +56,12 @@ auto selectGeneric(Func&& f, common::IterableRange2D<SizeType, LocalTile_TAG> ra
 template <class T, Device device>
 class Matrix : public Matrix<const T, device> {
 public:
+  static constexpr Device D = device;
```

Comment:
I know I've made this same change earlier in `Tile`, but now I'm wondering if the names should be swapped like this:
```suggestion
template <class T, Device D>
class Matrix : public Matrix<const T, D> {
public:
  static constexpr Device device = D;
```
(and of course with the same change in `Tile`, and now `Panel`). I think more often than not we template as `template <class T, Device D>` (and it fits with template parameters being CapitalCase), and variables are usually snake_case (like `coord` in `Panel`).

Absolutely not a blocker, but wanted to mention it here since you have it as one of the points in the description.

Reply:
If @rasolca agree on your proposal, I will quickly create a PR for that 😉

(I still haven't checked if I really need it in this PR, but I think it can be generally useful to be able to access this information.)

Reply:
I opened #685 trying to address the minimal set that would be useful for this.

As soon as that one will be merged, I will rebase this one and eventually do the last changes here if needed (i.e. checking if adding `device` attribute to `Matrix` and `Panel` is actually needed)